lib/bowermanager.js

```diff
@@ -0,0 +1,338 @@
+                    if (options.strict) {
+                      return callback(err);
+                    }
+        
+                    logger.log('error', 'Failed to add package: ' + pluginString, err);
+                    return callback(null);
+                  }
+          
+                  logger.log('info', 'Added package: ' + pluginString);
+          
+                  // Retrieve any older versions of the plugin.
+                  db.retrieve(plugin.getPluginType(), { name: package.name, version: { $ne: newPlugin.version } }, function (err, results) {
+                    if (err) {
+                      // strictness doesn't matter at this point
+                      logger.log('error', 'Failed to retrieve previous packages: ' + err.message, err);
+                    }
+          
+                    // Remove older versions of this plugin.
+                    db.destroy(plugin.getPluginType(), { name: package.name, version: { $ne: newPlugin.version } }, function (err) { 
+                      if (err) {
+                        logger.log('error', err);
+                        return callback(err);
+                      }
+      
+                      logger.log('info', 'Successfully removed versions of ' + package.name + ' (' + plugin.getPluginType() + ') older than ' + newPlugin.version);
+      
+                      return callback(null, newPlugin);
+                    }); // Remove older versions of the plugin.
+                  }); // Retrieve older versions of the plugin.
+                }); // Add the new plugin.
+              }); // Check if a plugin with this name and version already exists.
+            });
+          });
+        });
+      });
+    });
+  });
+}
+
+exports = module.exports = {
+
+  // expose the bower manager constructor
+  BowerManager : BowerManager,
+
+  /**
+   * preload function
+   *
+   * @param {object} app - AdaptBuilder instance
+   * @return {object} preloader - an AdaptBuilder ModulePreloader
+   */
+  preload : function (app) {
+    var preloader = new app.ModulePreloader(app, MODNAME, { events: this.preloadHandle(app, new BowerManager()) });
+    return preloader;
+  },
+
+  /**
+   * Event handler for preload events
+   *
+   * @param {object} app - Server instance
+   * @param {object} instance - Instance of this module
+   * @return {object} hash map of events and handlers
+   */
+  preloadHandle : function (app, instance){
+
+    return {
+        preload : function(){
+          var preloader = this;
+          preloader.emit('preloadChange', MODNAME, app.preloadConstants.WAITING);
+        },
+
+        moduleLoaded : function(modloaded){
+          var preloader = this;
+
+           //is the module that loaded this modules requirement
+          if (modloaded === WAITFOR) {
+            app.bowermanager = instance;
+            preloader.emit('preloadChange', MODNAME, app.preloadConstants.COMPLETE);
+          }
+        }
+      };
+  }
+};
```

Comment:
In time, yes.  I was going to come back to that  This would impact the plugin upload functionality and the initial install, etc. which is beyond what I was originally trying to resolve.